proxylib/aws_sqs/aws-sqs-parser.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsparsers
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"regexp"
+
+	. "github.com/cilium/cilium/proxylib/proxylib"
+
+	"github.com/cilium/proxy/go/cilium/api"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"net/url"
+)
+
+//
+// AWS Simple Queuing Service (SQS) Parser
+//
+// Spec: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-making-api-requests.html
+//       https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/Welcome.html
+//
+
+// Current AWS SQS parser supports matching on 'Action' and 'QueueName', both of which can be wildcarded.
+// QueueName can be a regular expression.  Per the AWS spec, queue names are case sensitive.
+// Examples:
+// action = 'CreateQueue', queue_name = 'MyQueue'
+
+type AWSSQSRule struct {
+	actionExact            string
+	queueNameRegexCompiled *regexp.Regexp
+}
+
+type AWSSQSRequestData struct {
+	action    string
+	queueName string
+}
+
+var AccessDeniedStr string = "HTTP/1.1 403 Access Denied\r\n\r\n"
+
+func (rule *AWSSQSRule) Matches(data interface{}) bool {
+	// Cast 'data' to the type we give to 'Matches()'
+
+	reqData, ok := data.(AWSSQSRequestData)
+	if !ok {
+		log.Warning("Matches() called with type other than string")
+		return false
+	}
+	log.Infof("Match Request: action '%s' queueName '%s'", reqData.action, reqData.queueName)
```

Comment:
I guess this will be printed for every request. Maybe making it Debug?

Reply:
Yes, @aanm , sorry, I should have made a note of this.  Right now the unit test framework hides items at the Debugf level, so anything i want to see while doing development I put at Infof.  I'll then downgrade them to Debugf before doing a version that is open to merging.   I'm not sure if I'm just missing something, or if this is something we should change about unit tests.  @jrajahalme  might now if I'm missing some way of seeing Debugf output in unit tests.